libs/user-identity-service/src/main/java/uk/gov/di/ipv/core/library/service/UserIdentityService.java

```diff
@@ -507,61 +502,50 @@ private List<ReturnCode> mapCisToReturnCodes(ContraIndicators contraIndicators)
                 .toList();
     }
 
-    private JsonNode getVcClaimNode(String credential, String node)
-            throws CredentialParseException {
-        try {
-            return OBJECT_MAPPER
-                    .readTree(SignedJWT.parse(credential).getPayload().toString())
-                    .path(VC_CLAIM)
-                    .path(node);
-        } catch (JsonProcessingException | ParseException e) {
-            throw new CredentialParseException(
-                    "Encountered a parsing error while attempting to parse VC store item: "
-                            + e.getMessage());
-        }
+    private List<NameParts> getNamePartsFromCredentialSubjectName(uk.gov.di.model.Name name) {
+        return name.getNameParts().stream()
+                .map(np -> new NameParts(np.getValue(), np.getType().toString()))
+                .toList();
     }
 
-    private <T> T getJsonProperty(JsonNode jsonNode, String propertyName, CollectionType valueType)
+    private IdentityClaim getIdentityClaim(VerifiableCredential vc)
             throws HttpResponseExceptionWithErrorBody {
-        JsonNode propertyNode = jsonNode.path(propertyName);
-        if (propertyNode.isMissingNode()) {
-            return OBJECT_MAPPER.convertValue(List.of(), valueType);
-        }
-        try {
-            return OBJECT_MAPPER.treeToValue(propertyNode, valueType);
-        } catch (JsonProcessingException e) {
-            LOGGER.error(LogHelper.buildErrorMessage("Failed to parse VC JWT", e));
+        if (vc.getCredential().getCredentialSubject() instanceof PersonWithIdentity person) {
+
+            List<Name> names =
+                    person.getName() != null
+                            ? person.getName().stream()
+                                    .map(
+                                            name ->
+                                                    new Name(
+                                                            getNamePartsFromCredentialSubjectName(
+                                                                    name)))
+                                    .toList()
+                            : List.of();
+
+            List<BirthDate> birthDates =
+                    person.getBirthDate() != null
+                            ? person.getBirthDate().stream()
+                                    .map(bd -> new BirthDate(bd.getValue()))
+                                    .toList()
+                            : List.of();
+
+            return new IdentityClaim(names, birthDates);
+
+        } else {
+            LOGGER.error(
+                    LogHelper.buildLogMessage(
+                            "Credential subject must be of type PersonWithIdentity."));
             throw new HttpResponseExceptionWithErrorBody(
                     500, ErrorResponse.FAILED_TO_GENERATE_IDENTIY_CLAIM);
```

Comment:
I'm not sure if this should be an error actually - will we definitely only pass credentials with an appropriate identity to this method?

It might be more consistent to just return an `IdentityClaim` with an empty list of names/birthDates?

Reply:
ah you're right! Address is passed in too (though it feels like if this method is only for getting an identity claim it should only be used for VCs with an identity claim? This seems like a not-so obvious side-effect and maybe worth a refactor in future?)

Reply:
Yeah it feels a bit odd - perhaps it should return `Optional.empty()` or `null` for non-identity VCs to make it a bit clearer - or force the caller to pass in an `IdentityCheckCredential`.

Perhaps better left as a separate refactor?